pyfritzhome/fritzhome.py

```diff
@@ -200,8 +201,17 @@ def _get_listinfo_elements(self, entity_type):
     def wait_device_txbusy(self, ain, retries=10):
         """Wait for device to finish command execution."""
         for _ in range(retries):
-            plain = self.get_device_infos(ain)
-            dom = ElementTree.fromstring(plain)
+            if self._has_getdeviceinfos:
+                try:
+                    plain = self.get_device_infos(ain)
+                    dom = ElementTree.fromstring(plain)
+                except exceptions.HTTPError:
```

Comment:
Hi @mib1185,
I think we should check for the status code here, to avoid sending more requests to the box in case of other errors 🙂.

Reply:
I remember getting 503 from a Fritzbox while testing something, but I don't remember how that happened 😉.